auth/src/auth/auth.controller.ts

```diff
@@ -0,0 +1,60 @@
+import {
+  Controller,
+  Body,
+  Post,
+  UseInterceptors,
+  ClassSerializerInterceptor,
+  SerializeOptions,
+  HttpCode,
+  HttpStatus,
+  Get,
+  Req,
+  UseGuards,
+} from '@nestjs/common';
+import { plainToInstance } from 'class-transformer';
+import { UserEntity } from '../db/entities/users.entity';
+import { AuthService } from './auth.service';
+import { LoginDTO } from './dtos/login.dto';
+import { SignUpDTO } from './dtos/signup.dto';
+import { RefreshTokenGuard } from './guards/refresh-token.guard';
+import { RequestWithRefreshPayload } from './interfaces/request-with-refresh-payload.interface';
+import { TokenResponse } from './responses/token.response';
+
+@Controller('auth')
+@UseInterceptors(ClassSerializerInterceptor)
+@SerializeOptions({ strategy: 'excludeAll' })
+export class AuthController {
+  constructor(private readonly authService: AuthService) {}
+
+  @Post('signup')
+  async signup(@Body() signUpData: SignUpDTO) {
+    return plainToInstance(
```

Comment:
There once was an issue with this fireing twice if you combine it with the ClassSerializerInterceptor.
Please test if that is the case.
And if you can just return without the plainToInstance since the signup already Promises the UserEntity Type and it may be handled that way already, or if best practice is usingo "new Objectname(...)"

Reply:
True the plainToInstance was not necessary and I went with just returning the result of the function. Creating a new Object was unnecessary, due to it already being the object it needed to be.